src/main/java/org/veupathdb/service/eda/ds/plugin/pass/TablePlugin.java

```diff
@@ -101,10 +105,30 @@ protected void writeResults(OutputStream out, Map<String, InputStream> dataStrea
     
     //loop through and print data
     first = true;
+    int offsetCount = 0;
+    int rowCount = 0;
     while (s.hasNextLine()) {
-      String[] row = parser.parseLineToArray(s.nextLine());
-      if (first) first = false; else out.write(",".getBytes());
-      out.write(new JSONArray(row).toString().getBytes());
+      if (offset != null) {
+        if (offsetCount < offset) {
+          offsetCount++;
+          s.nextLine();
+        } else {
+          if (numRows != null) {
+            if (rowCount < numRows) {
+              rowCount++;
+              String[] row = parser.parseLineToArray(s.nextLine());
+              if (first) first = false; else out.write(",".getBytes());
+              out.write(new JSONArray(row).toString().getBytes());
```

Comment:
To avoid using getBytes() everywhere and (more importantly) improve performance, wrap the passed OutputStream in a BufferedWriter and write to that instead.  e.g.
BufferedWriter bufOut = new BufferedWriter(new OutputStreamWriter(out));
bufOut.write(someString);